### ISSUE TITLE
Add new decoder and rules for HAProxy to detect brute force attack

### ISSUE THREAD

**Issue by josema0097** [wazuh/wazuh-ruleset#588]
Hello team,

## Introduction
The main goal of this issue is to create new alerts when we detect a brute force attack in the logs of HAProxy. This use case is not included in our current ruleset, so it would be great to share it with the rest of users

## Design

This is an example of a decoder to fetch the source ip from the HAproxy log:
```xml
<decoder name="haproxy">
  <program_name>^haproxy</program_name>
</decoder>

<decoder name="haproxy">
  <parent>haproxy</parent>
  <regex>(\d+.\d+.\d+.\d+)</regex>
  <order>srcip</order>
</decoder>
```

Furthermore these rules are an example of brute force attack detection when at least 8 attempts are made in less than a 30 seconds interval:
```xml
        <rule id="100011" level="3">
                <program_name>haproxy</program_name>
                <description>srcip</description>
        </rule>

        <rule id="100012" level="8" frequency="8" timeframe="30">
                <if_matched_sid>100011</if_matched_sid>
                <same_source_ip />
                <description>CMS (WordPress or Joomla) brute force attempt.</description>
                <group>pci_dss_6.5,pci_dss_11.4,pci_dss_6.5.10,pci_dss_10.2.4,pci_dss_10.2.5,gdpr_IV_35.7.d,gdpr_IV_32.2,hipaa_164.312.b,nist_800_53_SA.11,nist_800_53_SI.4,nist_800_53_AU.14,nist_800_53_AC.7,</group>
        </rule>
```

## References 
- https://groups.google.com/forum/#!topic/wazuh/RqTp-5rfQT0 (thanks Giorgio for your contribution)

Regards.

**Comment by iasdeoupxe** [comment]
https://github.com/wazuh/wazuh/pull/4338 is a contribution by @gkissand for a haproxy decoder but unfortunately ended up in the wrong repository and the MR doesn't move forward since a few month.

This could be a chance to pull the decoder from there into this repository and improve / extend it with the suggestions above.

**Comment by danielmjmarques** [comment]
I created these rules and decoders for Haproxy. Based on what I found on the Internet. I hope it helps someone else.
[Haproxy_Wazuh.zip](https://github.com/wazuh/wazuh-ruleset/files/6258995/Haproxy_Wazuh.zip)

**Comment by iasdeoupxe** [comment]
@danielmjmarques Looks like there is already an open merge request for decoders available by @palaciosjeremias at #739 Could make sense if your suggestion could be incorporated into this.

**Comment by jcruzlp** [comment]
Changes added here: https://github.com/wazuh/wazuh/pull/9033
